nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,280 @@
+        .identifiesControllerService(RecordReaderFactory.class)
+        .addValidator(Validator.VALID)
+        .required(true)
+        .build();
+
+    static final PropertyDescriptor RECORD_PATH = new PropertyDescriptor.Builder()
+        .name("rest-lookup-record-path")
+        .displayName("Record Path")
+        .description("An optional record path that can be used to define where in a record to get the real data to merge " +
+                "into the record set to be enriched. See documentation for examples of when this might be useful.")
+        .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+        .addValidator(Validator.VALID)
+        .required(false)
+        .build();
+
+    static final PropertyDescriptor SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
+            .name("rest-lookup-ssl-context-service")
+            .displayName("SSL Context Service")
+            .description("The SSL Context Service used to provide client certificate information for TLS/SSL "
+                    + "connections.")
+            .required(false)
+            .identifiesControllerService(SSLContextService.class)
+            .build();
+
+    private static final ProxySpec[] PROXY_SPECS = {ProxySpec.HTTP_AUTH, ProxySpec.SOCKS};
+    public static final PropertyDescriptor PROXY_CONFIGURATION_SERVICE
+            = ProxyConfiguration.createProxyConfigPropertyDescriptor(true, PROXY_SPECS);
+
+    static final String ENDPOINT_KEY = "endpoint";
+    static final String MIME_TYPE_KEY = "mime.type";
+    static final String BODY_KEY = "request.body";
+    static final String METHOD_KEY = "request.method";
+
+    static final List<PropertyDescriptor> DESCRIPTORS;
+
+    static {
+        DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            RECORD_READER,
+            RECORD_PATH,
+            SSL_CONTEXT_SERVICE,
+            PROXY_CONFIGURATION_SERVICE
+        ));
+    }
+
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    private ProxyConfigurationService proxyConfigurationService;
+    private SSLContextService sslContextService;
+    private RecordReaderFactory readerFactory;
+    private RecordPath recordPath;
+    private OkHttpClient client;
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        readerFactory = context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+        proxyConfigurationService = context.getProperty(PROXY_CONFIGURATION_SERVICE)
+                .asControllerService(ProxyConfigurationService.class);
+
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+
+        if (proxyConfigurationService != null) {
+            setProxy(builder);
+        }
+
```

Comment:
why are we getting the SSL_CONTEXT_SERVICE twice?

Reply:
Copy pasta